pkg/artifact.go

```diff
@@ -15,7 +16,21 @@ func LoadArtifactFromFile(filename string, mediaType string) (*ocispec.Descripto
 		return nil, nil, fmt.Errorf("error loading artifact from file: %w", err)
 	}
 
-	return LoadArtifactFromReader(file, mediaType)
+	desc, artifactBytes, err := LoadArtifactFromReader(file, mediaType)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Check if the descriptor already has a title annotation, if not, add it
+	if desc.Annotations == nil || desc.Annotations[ocispec.AnnotationTitle] == "" {
+		if desc.Annotations == nil {
+			desc.Annotations = make(map[string]string)
+		}
```

Comment:
We could have a separate function that handles this logic and write unit tests for this specifically. Then the unit testing becomes more lightweight in `Load...FromFile`